test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/BodyHeadTagHelperTest.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    public class BodyHeadTagHelperTest
```

Comment:
This should be `TagHelperComponentTagHelperTest` same with file name. In the underlying tests use a test class to verify the abstraction. Your functional tests verify the Body/Head variants of it.